### ISSUE TITLE
Error formatting a line ending with a closing paren when followed by a line starting with private[scope]

### ISSUE THREAD

**Issue by jebarb** [scalameta/scalafmt#1468]
This template is a guideline, not a strict requirement.
- **Version**: 2.0.0
- **Integration**: cli
- **Configuration**:
```
version = "2.0.0"
project.git = true
align=none
maxColumn = 120
continuationIndent.defnSite = 2
assumeStandardLibraryStripMargin = true
newlines.alwaysBeforeTopLevelStatements = true
verticalMultiline.atDefnSite = true
verticalMultiline.arityThreshold = 1
verticalMultiline.newlineAfterOpenParen = true
rewrite.sortModifiers.order = [
 "override", "private", "protected", "implicit", "final", "sealed", "abstract", "lazy"
]
rewrite.rules = [AvoidInfix, PreferCurlyFors, RedundantBraces, RedundantParens, SortImports, SortModifiers]
```

## Steps

Given code like this:
```scala
case class Foo(val foo: String)
private[example] case object Bar
```

When I run scalafmt like this:

```bash
scalafmt
```

## Problem

Scalafmt fails with the following error and stack trace:
```scala
org.scalafmt.cli.FailedToFormat: <path/to/class.scala>
Caused by: org.scalafmt.Error$PreciseIncomplete: <path-to-class.scala>:<line>: error: Unable to format file due to bug in scalafmt
private[example] case class Foo(val foo: String)
                                               ^
	at org.scalafmt.Scalafmt$.format(Scalafmt.scala:76)
	at org.scalafmt.cli.ScalafmtCoreRunner$.unsafeHandleFile(ScalafmtCoreRunner.scala:86)
	at org.scalafmt.cli.ScalafmtCoreRunner$.handleFile(ScalafmtCoreRunner.scala:65)
	at org.scalafmt.cli.ScalafmtCoreRunner$.$anonfun$run$3(ScalafmtCoreRunner.scala:43)
	at org.scalafmt.cli.ScalafmtCoreRunner$.$anonfun$run$3$adapted(ScalafmtCoreRunner.scala:42)
	at scala.collection.parallel.mutable.ParArray$ParArrayIterator.foreach(ParArray.scala:145)
	at scala.collection.parallel.ParIterableLike$Foreach.leaf(ParIterableLike.scala:974)
	at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:53)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:67)
	at scala.collection.parallel.Task.tryLeaf(Tasks.scala:56)
	at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:50)
	at scala.collection.parallel.ParIterableLike$Foreach.tryLeaf(ParIterableLike.scala:971)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.internal(Tasks.scala:170)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.internal$(Tasks.scala:157)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.internal(Tasks.scala:440)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute(Tasks.scala:150)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute$(Tasks.scala:149)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:440)
	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:473)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
```

## Workaround
Adding a comment before the second line prevents the failure and allows files to be formatted. The following can be successfully formatted:
```scala
case class Foo(val foo: String)
// fixed
private[example] case object Bar
```

## Notes

This seems to fail every time a line starting with with ```private[scope]``` is preceded by a line ending with a closing paren.

**Comment by tanishiking** [comment]
Maybe this has fixed with https://github.com/scalameta/scalafmt/pull/1513 which is released as scalafmt 2.1.0 ?
Could you give it a try on the latest version of scalafmt? @jebarb

**Comment by jebarb** [comment]
@tanishiking I'm using https://github.com/mroth/scalafmt-native so I'll need to wait for a new release before testing, but based on the added test case it's fixed
